### PR TITLE
fix(desk-tool): fix ui and performance issues in timeline

### PR DESCRIPTION
**Pull request by hermanwikner**
### Description

This PR adds UI and performance improvements to the timeline.

- Render items in a `VirtualList` to prevent app from freezing when there is a lot of data to display
- Improve 'scroll-to-load-more'
- Update the styling of different states / interactions
- General updates to improve performance such as memoization

### CODE REVIEW

File: packages/@sanity/desk-tool/src/panes/document/timeline/__workshop__/TimelineItemStory.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import {Card, Container, Flex, LayerProvider, Menu} from '@sanity/ui'
```
Comment (RitaDias) [inline packages/@sanity/desk-tool/src/panes/document/timeline/__workshop__/TimelineItemStory.tsx:1]: What's our stance on adding things to the workshop and push them to next? Should we keep it as a dev tool or keep it as well and push it to next? 🤔

File: packages/@sanity/desk-tool/src/panes/document/timeline/timelineItem.styled.tsx [hunk 2]
```diff
@@ -12,104 +12,99 @@ export interface TimelineItemProps {
...
       }
+    }
 
-      ${IconWrapper} {
-        &::before {
-          background: var(--card-hairline-soft-color);
-        }
+    &:not([disabled], [data-selected]) {
+      &[data-active='true'] {
+        --card-fg-color: ${theme.sanity.color.selectable?.primary.selected.fg};
```
Comment (RitaDias) [inline packages/@sanity/desk-tool/src/panes/document/timeline/timelineItem.styled.tsx:36]: This colour seems to be used a lot all the way down 👇 maybe it makes sense to just extract it to a const above all of this and be called in each of these properties instead?

like
`const selectedColor = theme.sanity.color.selectable?.primary.selected.fg`
(and so on with the other that might be repeated)

and then here you'd just `--card-fg-color: ${selectedColor};`

File: packages/@sanity/desk-tool/src/panes/document/timeline/timelineItem.styled.tsx [hunk 3]
```diff
@@ -12,104 +12,99 @@ export interface TimelineItemProps {
...
+      &[data-selection-within='true'] {
         border-radius: 0;
       }
+    }
 
-      ${IconWrapper} {
-        &::before {
-          background: var(--card-hairline-soft-color);
-        }
+    &:not([disabled], [data-selected]) {
```
Comment (RitaDias) [inline packages/@sanity/desk-tool/src/panes/document/timeline/timelineItem.styled.tsx:34]: I know that the names tend to be pretty direct and straightforward but I'd suggest adding comments to the top of these sections explaining what they are

File: packages/@sanity/desk-tool/src/panes/document/timeline/timelineItem.styled.tsx [hunk 4]
```diff
@@ -12,104 +12,99 @@ export interface TimelineItemProps {
...
+  `
+})
 
-    ${(isHovered || state === 'selected') &&
-    css`
-      ${IconWrapper}::before {
-        background: transparent;
+export const IconTimelineFlex = styled(Flex)(() => {
+  const line = css`
+    content: '';
```
Comment (RitaDias) [inline packages/@sanity/desk-tool/src/panes/document/timeline/timelineItem.styled.tsx:80]: Does this need the `content` when it's not a pseudo-element? 🤔

File: packages/@sanity/desk-tool/src/panes/document/timeline/timelineMenu.tsx [hunk 5]
```diff
@@ -55,6 +55,12 @@ export function TimelineMenu({chunk, mode}: TimelineMenuProps) {
   }, [handleClose])
 
   useClickOutside(handleClickOutside, [menuContent, buttonRef])
+  useGlobalKeyDown((e) => {
```
Comment (RitaDias) [inline packages/@sanity/desk-tool/src/panes/document/timeline/timelineMenu.tsx:58]: nice 😚 👌

File: packages/@sanity/desk-tool/src/panes/document/timeline/types.ts [hunk 6]
```diff
@@ -1 +1 @@
-export type TimelineItemState = 'enabled' | 'disabled' | 'withinSelection' | 'selected'
+export type TimelineItemState = 'enabled' | 'disabled' | 'withinSelection' | 'active'
```
Comment (RitaDias) [inline packages/@sanity/desk-tool/src/panes/document/timeline/types.ts:1]: Why, is it just about clarifying the name? 🤔

**Review by RitaDias — CHANGES_REQUESTED** [review]
Really good job 👏 I've added just a couple of questions and suggestion so just have a look ✨

File: packages/@sanity/desk-tool/src/panes/document/timeline/userAvatarStack.tsx [hunk 7]
```diff
@@ -1,20 +1,25 @@
...
+const AvatarCard = styled(Card)`
+  background: transparent;
+`
+
 export function UserAvatarStack({maxLength, userIds}: UserAvatarStackProps) {
   return (
     <AvatarStack maxLength={maxLength}>
       {userIds.map((userId) => (
-        <UserAvatar key={userId} userId={userId} withTooltip />
+        <AvatarCard key={userId}>
```
Comment (RitaDias) [inline packages/@sanity/desk-tool/src/panes/document/timeline/userAvatarStack.tsx:20]: Just out of pure curiosity, why was it needed to add this?

**Comment by mariuslundgard** [comment]
@hermanwikner – can you setup a workshop that shows the problems with using `VirtualList` and other aspects that remain to be sorted out?

**Pull request closed without merging**